server/metadata.go

```diff
@@ -1323,3 +1312,39 @@ func ensureTimeout(ctx context.Context, defaultTimeout time.Duration) (context.C
 	}
 	return ctx, cancel
 }
+
+// eventTimestampsToProto returns a client proto's partition event timestamps
+// from a partition's event timestamps struct.
+func eventTimestampsToProto(timestamps EventTimestamps) *client.PartitionEventTimestamps {
+	first, latest := timestamps.firstTime, timestamps.lastTime
+	result := &client.PartitionEventTimestamps{}
+
+	// Calling UnixNano() on a zero time is undefined, so we need to make these
+	// checks.
+	if !first.IsZero() {
+		result.FirstTimestamp = first.UnixNano()
+	}
+	if !latest.IsZero() {
+		result.LatestTimestamp = latest.UnixNano()
+	}
+
+	return result
+}
+
+// getPartitionMetadata returns a partition's metadata.
+func getPartitionMetadata(partitionId int32, partition *partition) *client.PartitionMetadata {
```

Comment:
I guess that it may be nice to have a separate test for this ?

Kind of : 

```golang
	p, err := server.newPartition(&proto.Partition{
		Subject:  "foo",
		Stream:   "foo",
		Replicas: []string{"a", "b"},
		Leader:   "a",
		Isr:      []string{"a", "b"},
	}, false, nil)
        // Mock MessagesReceivedTimestamps here
       partitionMetadata := getPartitionMetadata(0, partition)
       // assert everything is still there
       // id , leader, MessageTimestamps?
```

I think it would be really a unit test case then. As for higher order test case, I think `TestFetchPartitionMetadata` may cover already

Reply:
Not sure how much value a test for such a simple function would add: this function is only returning a new instance of a struct with its fields initialized using the partition, it doesn't do any processing at all.